tests/kademlia.rs

```diff
@@ -1,51 +1,99 @@
+        bootstrappers.push(Node::new().await);
     }
 
     // register the bootstrappers' ids and addresses
-    let mut peers = Vec::with_capacity(PEER_COUNT);
-    for node in &nodes {
-        let (id, addrs) = node.identity().await.unwrap();
+    let mut bootstrapper_ids = Vec::with_capacity(BOOTSTRAPPER_COUNT);
+    for bootstrapper in &bootstrappers {
+        let (id, addrs) = bootstrapper.identity().await.unwrap();
         let id = PeerId::from_public_key(id);
 
-        peers.push((id, addrs));
+        bootstrapper_ids.push((id, addrs));
     }
 
     // connect all the bootstrappers to one another
-    for (i, (node_id, _)) in peers.iter().enumerate() {
-        for (peer_id, addrs) in peers.iter().filter(|(peer_id, _)| peer_id != node_id) {
-            nodes[i]
-                .add_peer(peer_id.clone(), addrs[0].clone())
+    for (i, (node_id, _)) in bootstrapper_ids.iter().enumerate() {
+        for (bootstrapper_id, addrs) in bootstrapper_ids
+            .iter()
+            .filter(|(peer_id, _)| peer_id != node_id)
+        {
+            bootstrappers[i]
+                .add_peer(bootstrapper_id.clone(), addrs[0].clone())
                 .await
                 .unwrap();
         }
     }
 
-    // introduce an extra peer and connect it to one of the bootstrappers
-    let extra_peer = Node::new().await;
-    assert!(extra_peer
-        .add_peer(peers[0].0.clone(), peers[0].1[0].clone())
+    // introduce a peer and connect it to one of the bootstrappers
+    let peer = Node::new().await;
+    assert!(peer
+        .add_peer(
+            bootstrapper_ids[0].0.clone(),
+            bootstrapper_ids[0].1[0].clone()
+        )
+        .await
+        .is_ok());
+
+    // check that kad::bootstrap works
+    assert!(peer.bootstrap().await.is_ok());
+
+    // check that kad::get_closest_peers works
+    assert!(peer.get_closest_peers().await.is_ok());
+}
+
+#[async_std::test]
+async fn kademlia_popular_content_discovery() {
+    // set up logging
+    let _ = init_test_logging();
+
+    let (bootstrapper_id, bootstrapper_addr): (PeerId, Multiaddr) = (
+        "QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ"
+            .parse()
+            .unwrap(),
+        "/ip4/104.131.131.82/tcp/4001".parse().unwrap(),
+    );
+
+    // introduce a peer and specify the Kademlia protocol to it
+    // without a specified protocol, the test will not complete
+    let mut opts = IpfsOptions::inmemory_with_generated_keys();
+    opts.kad_protocol = Some("/ipfs/lan/kad/1.0.0".to_owned());
+    let peer = Node::with_options(opts).await;
+
+    // connect it to one of the well-known bootstrappers
+    assert!(peer
+        .add_peer(bootstrapper_id, bootstrapper_addr)
         .await
         .is_ok());
 
-    // call kad::bootstrap
-    assert!(extra_peer.bootstrap().await.is_ok());
+    // the Cid of the docs.ipfs.io website
+    let cid: Cid = "bafybeicfjz7woevc5dxvsskibxpxpofkrdjyslbggvvr3d66ddqu744nne"
```

Comment:
When this breaks, we should fix to the logo svg, which might be a bit more stable: bafkreicncneocapbypwwe3gl47bzvr3pkpxmmobzn7zr2iaz67df4kjeiq

Reply:
I actually did have it set to the svg at a point, must have accidentally reverted that bit (I was doing some tinkering around it)

Reply:
done :+1: